kernel/arch/i386/boot/bootparse.cpp

```diff
@@ -116,10 +123,166 @@ void px_parse_multiboot2(void *info)
             break;
         }
         default:
-            px_rs232_printf("Unknown Multiboot2 Tag: %d\n", tag->type);
+            px_rs232_printf("Unknown Multiboot2 tag: %d\n", tag->type);
             break;
         }
         // move to the next tag, aligning if necessary
         tag = (struct multiboot_tag*)(((uintptr_t)tag + tag->size + 7) & ~((uintptr_t)7));
     }
 }
+
+/*
+ *  ___ _   _          _     ___
+ * / __| |_(_)_ ____ _| |___|_  )
+ * \__ \  _| \ V / _` | / -_)/ /
+ * |___/\__|_|\_/\__,_|_\___/___|
+ */
+
+static void print_stivale2_mmap(struct stivale2_struct_tag_memmap* mmap)
+{
+    static const char *mmap_types[] = {
+        [0] = "Invalid",
+        [STIVALE2_MMAP_USABLE] = "Available",
+        [STIVALE2_MMAP_RESERVED] = "Reserved",
+        [STIVALE2_MMAP_ACPI_RECLAIMABLE] = "ACPI reclaimable",
+        [STIVALE2_MMAP_ACPI_NVS] = "Non-volatile storage",
+        [STIVALE2_MMAP_BAD_MEMORY] = "Bad RAM"
+    };
+    // List of memory map entries
+    auto mmap_list = (struct stivale2_mmap_entry*)mmap->memmap;
+    for (uint64_t i = 0; i < mmap->entries; i++)
+    {
+        auto entry = &mmap_list[i];
+        px_rs232_printf("  addr: 0x%02x%08x, length: 0x%02x%08x, type: %s\n",
+            (uint32_t)(entry->base >> 32) & 0xff, (uint32_t)(entry->base & UINT32_MAX),
+            (uint32_t)(entry->length >> 32) & 0xff, (uint32_t)(entry->length & UINT32_MAX),
+            // Have to do this because the IDs aren't in an order that can be
+            // mapped into an array like Multiboot. The last two types start at
+            // 0x1000, so we have to do some ternary operator magic.
+            (entry->type < 6 ? mmap_types[entry->type]
+                             : (entry->type == STIVALE2_MMAP_BOOTLOADER_RECLAIMABLE ? "Bootloader"
```

Comment:
It might be better to just write a function that returns the correct string. The split between an array, and a ternary operator feels kinda weird.

Reply:
I know that in the grand scheme of things a function call isn't that much time, but that seems a bit overkill when like you said before, we're going to be replacing all of this eventually when we have a use for it. I personally think ternaries are great ways to inline selection between two options.

Reply:
But you're selecting between more than 2 values...

Reply:
I see. I forgot that the others were in the array. I can make a function that returns the type. I'll just make it a switch statement.

Reply:
Resolved in 5dfcb9d